code/TagField.php

```diff
@@ -173,4 +273,90 @@ public function saveInto(DataObjectInterface $record) {
 			$record->$name = implode(',', $values);
 		}
 	}
+
+	/**
+	 * Returns a JSON string of tags, for ajax-based search.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return string
+	 */
+	public function suggest(SS_HTTPRequest $request) {
+		if($this->recordClass === null) {
+			return '[]';
+		}
+
+		/**
+		 * @var DataObject $object
+		 */
+		$object = singleton($this->recordClass);
```

Comment:
Why do we need this inline declaration? `singleton()` should already provide that level of type hinting. I also don't like block level comments inside methods, makes it harder to comment out stuff.